server/src/main/java/org/elasticsearch/common/Strings.java

```diff
@@ -286,11 +292,22 @@ private static String changeFirstCharacterCase(String str, boolean capitalize) {
         .collect(Collectors.joining(",", "[", "]"));
 
     public static boolean validFileName(String fileName) {
-        return fileName.chars().noneMatch(c -> isInvalidFileNameCharacter((char) c));
+        for (int i = 0; i < fileName.length(); i++) {
```

Comment:
This is hot in a couple of spots, the `validFileNameExcludingAstrix` isn't but I figured if I fix one I might as well fix both.